src/shared/Core/Trace2.cs

```diff
@@ -150,6 +190,51 @@ public interface ITrace2 : IDisposable
         string filePath = "",
         [System.Runtime.CompilerServices.CallerLineNumber]
         int lineNumber = 0);
+
+    Region CreateRegion(
+        string category,
+        string label,
+        [System.Runtime.CompilerServices.CallerFilePath]
+        string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber]
+        int lineNumber = 0,
+        string message = "");
+
+    /// <summary>
+    /// Writes a region enter message to the trace writer.
+    /// </summary>
+    /// <param name="category">Category of region.</param>
+    /// <param name="label">Description of region.</param>
+    /// <param name="threadName">Name of the currently-executing thread.</param>
+    /// <param name="message">Message associated with entering region.</param>
+    /// <param name="filePath">Path of the file this method is called from.</param>
+    /// <param name="lineNumber">Line number of file this method is called from.</param>
+    void WriteRegionEnter(
+        string category,
+        string label,
+        string threadName,
+        string message = "",
+        string filePath = "",
+        int lineNumber = 0);
```

Comment:
Are we're missing the `[System.Runtime.CompilerServices.CallerX]` attributes here?

Reply:
Well spotted.